src/gicv3.rs

```diff
@@ -67,6 +67,31 @@ pub struct GicV3<'a> {
     gicr_stride: usize,
 }
 
+fn get_redistributor_window_size(gicr_base: *mut GicrSgi, gic_v4: bool) -> usize {
+    if !gic_v4 {
+        return size_of::<GicrSgi>();
+    }
+
+    // SAFETY: The caller of `GicV3::new` promised that `gicr_base` was valid
+    // and there are no aliases.
+    let first_gicr_window = unsafe { UniqueMmioPointer::new(NonNull::new(gicr_base).unwrap()) };
+
+    // SAFETY: We only split out a single field.
+    let first_gicr = unsafe { split_fields!(first_gicr_window, gicr) };
```

Comment:
Can't you just use `field!` or `field_shared!` here, and avoid the unsafe block?